processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -268,6 +270,7 @@ public int lookupId(String name)
   private final Map<String, MetricDesc> metricDescs;
   private final Map<String, DimensionDesc> dimensionDescs;
   private final Map<String, ColumnCapabilitiesImpl> columnCapabilities;
+  private final DimensionValueHolder dimValues;
```

Comment:
it looks like that DimensionValueHolder is equivalent of `Collections.synchronizedList(Lists.newArrayListWithCapacity(Math.max(32, numDimension)))` , why introduce a whole new class?